File: scan-service/src/InspectionTag.php
<?php

declare(strict_types=1);

namespace VuuroScan;

final class InspectionTag
{
    public const VALUES = [
        'damage',
        'wear_and_tear',
        'missing_item',
        'safety_issue',
        'pre_existing_condition',
        'maintenance_needed',
        'confirmed_present',
        'other',
    ];

    public const LABELS = [
        'damage' => 'Damage',
        'wear_and_tear' => 'Wear and tear',
        'missing_item' => 'Missing item',
        'safety_issue' => 'Safety issue',
        'pre_existing_condition' => 'Pre-existing condition',
        'maintenance_needed' => 'Maintenance needed',
        'confirmed_present' => 'Confirmed present',
        'other' => 'Other',
    ];

    public static function isValidList(array $tags): bool
    {
        foreach ($tags as $tag) {
            if (!is_string($tag) || !in_array($tag, self::VALUES, true)) {
                return false;
            }
        }
        return true;
    }

    public static function labelFor(string $value): string
    {
        return self::LABELS[$value] ?? $value;
    }
}

File: ios-app/Sources/Models/InspectionTag.swift
import Foundation

enum InspectionTag: String, CaseIterable, Codable, Identifiable {
    case damage
    case wearAndTear = "wear_and_tear"
    case missingItem = "missing_item"
    case safetyIssue = "safety_issue"
    case preExistingCondition = "pre_existing_condition"
    case maintenanceNeeded = "maintenance_needed"
    case confirmedPresent = "confirmed_present"
    case other

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .damage: return "Damage"
        case .wearAndTear: return "Wear and tear"
        case .missingItem: return "Missing item"
        case .safetyIssue: return "Safety issue"
        case .preExistingCondition: return "Pre-existing condition"
        case .maintenanceNeeded: return "Maintenance needed"
        case .confirmedPresent: return "Confirmed present"
        case .other: return "Other"
        }
    }
}
